File: src/dataset/python/space-age_exercism-ThomasZumsteg/openAI/ChatGPT4_space_age_v1.py
"""Calculates age on various planets"""

class SpaceAge:
    """Calculates age on various planets"""
    _earth_year = 31557600  # seconds
    _ratios = {
        "on_earth": 1.0,
        "on_mercury": 0.2408467,
        "on_venus": 0.61519726,
        "on_mars": 1.8808158,
        "on_jupiter": 11.862615,
        "on_saturn": 29.447498,
        "on_uranus": 84.016846,
        "on_neptune": 164.79132,
    }

    def __init__(self, seconds):
        """Stores age and precomputes planet ages"""
        self.seconds = seconds
        self._ages = {
            name: round(self.seconds / (self._earth_year * ratio), 2)
            for name, ratio in self._ratios.items()
        }

    def __getattr__(self, name):
        """Dynamically returns precomputed planet age"""
        if name in self._ages:
            return lambda: self._ages[name]
        raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")

File: src/dataset/Java/space-age_exercism-ThomasZumsteg/openAI/ChatGPT4_SpaceAge_v1.java
public class SpaceAge {
    private final double seconds;
    private static final double EARTH_YEAR_SECONDS = 31557600.0;
    private static final double MERCURY_ORBIT = 0.2408467;
    private static final double VENUS_ORBIT = 0.61519726;
    private static final double MARS_ORBIT = 1.8808158;
    private static final double JUPITER_ORBIT = 11.862615;
    private static final double SATURN_ORBIT = 29.447498;
    private static final double URANUS_ORBIT = 84.016846;
    private static final double NEPTUNE_ORBIT = 164.79132;

    public SpaceAge(Long seconds) {
        this.seconds = seconds;
    }

    public SpaceAge(Integer seconds) {
        this.seconds = seconds.doubleValue();
    }

    public Long getSeconds() {
        return (long) this.seconds;
    }

    public double onEarth() {
        return this.seconds / EARTH_YEAR_SECONDS;
    }

    public double onMercury() {
        return this.seconds / (EARTH_YEAR_SECONDS * MERCURY_ORBIT);
    }

    public double onVenus() {
        return this.seconds / (EARTH_YEAR_SECONDS * VENUS_ORBIT);
    }

    public double onMars() {
        return this.seconds / (EARTH_YEAR_SECONDS * MARS_ORBIT);
    }

    public double onJupiter() {
        return this.seconds / (EARTH_YEAR_SECONDS * JUPITER_ORBIT);
    }

    public double onSaturn() {
        return this.seconds / (EARTH_YEAR_SECONDS * SATURN_ORBIT);
    }

    public double onUranus() {
        return this.seconds / (EARTH_YEAR_SECONDS * URANUS_ORBIT);
    }

    public double onNeptune() {
        return this.seconds / (EARTH_YEAR_SECONDS * NEPTUNE_ORBIT);
    }
}